catalyst-gateway/bin/src/service/api/documents/common/mod.rs

```diff
@@ -68,9 +68,14 @@ impl catalyst_signed_doc::providers::CatalystIdProvider for ValidationProvider {
         &self,
         kid: &catalyst_signed_doc::CatalystId,
     ) -> anyhow::Result<Option<ed25519_dalek::VerifyingKey>> {
-        self.verifying_key_provider
-            .try_get_registered_key(kid)
-            .await
+        let Some(admin_key) = get_admin_key(kid) else {
```

Comment:
Its not a secure and safe verification that the provided key is admin or not.
In your implementation if we cannot find an admin key we fall back to the original implementation, which would allow for the user to sign admin related documents and publish them.
The logic should be something like this:
```rust
if kid.is_admin() {
  get_admin_key(kid)
} else {
   self.verifying_key_provider
                .try_get_registered_key(kid)
                .await
}
``` 
So if the provided key defined as admin, we just looking into the admin keys registry.